src/Lecture1.hs

```diff
@@ -123,4 +142,17 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> [Char]
+lowerAndGreater n list =
+    let numGreater = count 0 list (\ x-> x<n)
+        numLesser = count 0 list (\ x-> x>n)
```

Comment:
That's correct! 💯 

However, with this solution, you will walk through the list twice.
You can write it more performant with only one traversal of the list 🙂
It's desirable to write functions that traverse lists only a single time and consume elements only once. In that case, the composition of such functions will have nice streaming and lazy properties.

Reply:
Thank you for all the comments!
I do have a question for this particular one: I now managed to traverse the list only once, but used a Tuple to pass on the results in the recursive function call. Is there an alternative to using a Tuple, which was only introduced in Lecture 2?

Reply:
Yes, it's possible to write the function differently. You can pass both numbers as separate arguments and make the function return `String`, for example. Something like this:

```haskell
go :: Int -> Int -> [Int] -> String
    go l h [] = show n ++ " is greater than " ++ ...
    go l h (x : xs)
      | x < n = go (l + 1) h xs
      | ...
```